util/prompt_builder.py

```diff
@@ -0,0 +1,87 @@
+"""Module to construct prompts for LLM calls."""
+
+from pathlib import Path
+from string import Template
+from subprocess import CompletedProcess
+
+from util import ParsecFunction, ParsecResult
+
+
+class PromptBuilder:
+    """Encapsulates functions used in constructing prompts for LLM calls."""
+
+    INITIAL_PROMPT_FOR_SPECIFICATION_GENERATION = Path("./prompt.txt").read_text()
+    SOURCE_DELIMITER = "<<SOURCE>>"
```

Comment:
I think of a pair of delimiters bracketing something (though you can also have one delimiter that separates two things).  Since this gets replaced, I would call it a "placeholder" rather than a "delimiter".